src/debug-adapter/debugEventManager.ts

```diff
@@ -111,7 +116,7 @@ export class DebugEventManager {
     panelViewStackFrame: PanelViewStackFrame
   ): Promise<void> {
     let isNewAndObject = false;
-    const id = variable.value;
+    const id = variable.value === 'null' ? variable.value : `${DebugEventManager.objectPrefix}${variable.value.split(this.sizeSuffix)[0]}`;
```

Comment:
Why is the split necessary?

Reply:
When an arrayList is given the variable.value looks something like this: 
'ArrayList@17 size=3'
The problem is that size changes over the course of the program which in turn means that every time an element is added the array gets a new id. To avoid this problem, I only use the string before the “size=” specification.